contrib/backport/backport.go

```diff
@@ -0,0 +1,443 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/urfave/cli"
+	"gopkg.in/yaml.v2"
+)
+
+const defaultVersion = "v1.18"
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "backport"
+	app.Usage = "Backport provided PR-number on to the current or previous released version"
+	app.Description = `Backport will look-up the PR in Gitea's git log and attempt to cherry-pick it on the current version`
+	app.ArgsUsage = "<PR-to-backport>"
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "version",
+			Usage: "Version branch to backport on to",
+		},
+		cli.StringFlag{
+			Name:  "upstream",
+			Value: "origin",
+			Usage: "Upstream remote for the Gitea upstream",
+		},
+		cli.StringFlag{
+			Name:  "release-branch",
+			Value: "",
+			Usage: "Release branch to backport on. Will default to <origin>/release/<version>",
+		},
+		cli.StringFlag{
+			Name:  "cherry-pick",
+			Usage: "SHA to cherry-pick as backport",
+		},
+		cli.StringFlag{
+			Name:  "backport-branch",
+			Usage: "backport branch to backport on to (default: backport-<pr>-<version>",
+		},
+		cli.StringFlag{
+			Name:  "remote",
+			Value: "",
+			Usage: "Remote for your fork of the Gitea upstream",
+		},
+		cli.StringFlag{
+			Name:  "fork-user",
+			Value: "",
+			Usage: "Forked user name on Github",
+		},
+		cli.BoolFlag{
+			Name:  "no-fetch",
+			Usage: "set this flag to prevent fetch of remote branches",
+		},
+		cli.BoolFlag{
+			Name:  "no-amend",
```

Comment:
Shouldn't this variable then be called `no-rewrite-commit-message`?

Reply:
It's amending the commit mesage using `git commit --amend`.

`--no-rewrite-commit-message` is extremely verbose and would be extremely difficult to get right when typing the command.

Reply:
Yep, that would be good.